src/codegen/trampoline.rs

```diff
@@ -62,7 +62,7 @@ pub fn generate(
 pub fn func_string(
     env: &Env,
     analysis: &Trampoline,
-    replace_self_bound: Option<impl AsRef<str>>,
+    replace_self_bound: &Option<impl AsRef<str>>,
```

Comment:
Not a fan, this was mostly for convenience since the contents of the option are typically a reference.

Reply:
I reverted that change. Is this fine for you or did you also mean some of the other functions?

Reply:
All occurrences please :)

Reply:
Okay, I removed a few more. Let me know if I removed all that you wanted me to revert

Reply:
If you are fine with it, those should have been the last requested changes and its good to go, right?

Reply:
I'm not a maintainer so that's not up to me to say.

One thing I'm missing here is the change that disallows these clippy lints across the crate. As you mentioned some are from the `pedantic` class but only few of these changes should have been necessary to make `clippy` on Rust 1.64 with the default/current configuration happy.
In other words, if you fixed additional lints that weren't even disallowed/warned about in the first place, please make sure they're disallowed now so that these issues don't slip right back in with the first next PR/contribution.

Reply:
@GuillaumeGomez what is your oppinion on this? Should all/some of them be added?

Reply:
CC @GuillaumeGomez @sdroege ^?

Reply:
I personally don't really see the point of this specific change. As for clippy overall, as I mentioned below, you can add more lints suites if you want, please just open an issue so it can be discussed beforehand.

Reply:
@GuillaumeGomez @sdroege Sorry, this thread isn't about the `&Option` change (which has been reverted for the same reason) _anymore_, but a reply to @pentamassiv's `If you are fine with it, those should have been the last requested changes and its good to go, right?`. It is instead about:

> For enabling new lints, that should be done on a case by case basis and of course also be enabled on the CI then.

Because yes, it's wasteful to @pentamassiv's time to address random lints but not make the CI fail on them in the future.

Reply:
As such, @pentamassiv please submit a PR that disallows all the lints you fixed here?